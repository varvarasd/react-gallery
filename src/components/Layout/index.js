import React from 'react';

import { Helmet } from 'react-helmet-async';

const Layout = ({links, title, description, children}) => {
    return <main className="main">
        <Helmet>
            {links}
            <title>{title}</title>
            <meta name="description" content={description} />
        </Helmet>
        {children}
    </main>
}

export default Layout;