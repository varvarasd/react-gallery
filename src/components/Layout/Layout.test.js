import React from 'react';
import { render } from '@testing-library/react';
import Layout from '.';
jest.mock('react-helmet-async');

describe('Layout', () => {
    it('renders', () => {
        const { container } = render(<Layout />);
        expect(container.firstChild).toMatchSnapshot();
    });

    it('renders children', () => {
        const { getByText } = render(<Layout><div>Layout child</div></Layout>);
        expect(getByText(/Layout child/)).toBeInTheDocument();
    });
});