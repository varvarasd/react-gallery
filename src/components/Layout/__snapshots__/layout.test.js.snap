// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders 1`] = `
<main
  class="main"
/>
`;
